services/connector/commands/sign.go

```diff
@@ -33,31 +39,40 @@ func (r *RPCRequest) getPersonalSignParams() (*PersonalSignParams, error) {
 		return nil, ErrInvalidParamsStructure
 	}
 
+	challengeIndex := 0
+	addressIndex := 1
+
+	if r.Method == Method_SignTypedDataV4 {
```

Comment:
Ok, since the request is always coming from the dapp, following this
https://docs.metamask.io/wallet/reference/json-rpc-methods/personal_sign/
https://docs.metamask.io/wallet/reference/json-rpc-methods/eth_signtypeddata_v4/
we're good with this condition.